lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -268,3 +280,90 @@ instance CompareDiscovery (RndState n) where
 
 instance KnownAddresses (RndState n) where
     knownAddresses s = Map.elems (addresses s)
+
+--------------------------------------------------------------------------------
+--
+-- RndAnyState
+--
+-- For benchmarking and testing arbitrary large random wallets.
+
+-- | An "unsound" alternative that can be used for benchmarking and stress
+-- testing. It re-uses the same underlying structure as the `RndState` but
```

Comment:
> same underlying structure

Is there a main difference, say in complexity, between `RndAnyState` and `AnyAddressState`?

Reply:
Would this be true?
```
RndAnyState, like RndState, grow with the number of addresses used. This is unlike SeqState/AnyAddressState, where which has a bounded gap of user addresses it keeps in memory
```
If not, something like it, would be cool to understand what makes `RndAnyState` indicative of performance of random wallets, and why it's not enough to have `AnyAddressState`.

Reply:
Ah, I'm blind 😅 
```haskell
innerState :: RndState network
```
"same underlying structure" makes sense then

Reply:
:) ... It still stores the UTxO and transaction history, but the `AnyState` do not store the address space. BUT, we've observed some major slowness when it comes to storing addresses, so I find it quite important to also include that in the benchmark. 

I think that we could have a similar `SeqAnyState` which also include the seq state, and get rid of the `AnyState` altogether.